Explorer/Assets/DCL/MapRenderer/ConsumerUtils/MapRenderImage.cs

```diff
@@ -73,32 +70,48 @@ public void OnPointerClick(PointerEventData eventData)
         {
             Profiler.BeginSample(POINTER_CLICK_SAMPLE_NAME);
 
-            //Process different click types if normal parcel or a map pin
-            if (isActive && !dragging && TryGetParcelUnderPointer(eventData, out Vector2Int parcel, out _, out _, out IPinMarker pinMarker))
+            if (isActive && !dragging && TryGetParcelUnderPointer(eventData, out Vector2Int parcel, out _, out _, out IPinMarker? pinMarker))
             {
-                if (pinMarker != null && previousClickedMarker != pinMarker)
-                {
-                    previousClickedMarker?.AnimateDeselectionAsync().Forget();
-                    pinMarker.AnimateSelectionAsync().Forget();
-                    previousClickedMarker = pinMarker;
-                }
-                else if (previousClickedMarker != null)
-                {
-                    previousClickedMarker.AnimateDeselectionAsync().Forget();
-                    previousClickedMarker = null;
-                }
-
-                ParcelClicked?.Invoke(new ParcelClickData
-                {
-                    Parcel = parcel,
-                    WorldPosition = GetParcelWorldPosition(parcel),
-                    PinMarker = pinMarker,
-                });
+                HandlePinMarkerClick(pinMarker);
+                InvokeParcelClicked(parcel, pinMarker);
             }
 
             Profiler.EndSample();
         }
 
+        public void OnSearchResultParcelSelected(Vector2Int parcel)
+        {
+            IPinMarker? pinMarker = interactivityController!.GetPinMarkerOnParcel(parcel);
+            HandlePinMarkerClick(pinMarker);
+            InvokeParcelClicked(parcel, pinMarker);
+        }
+
+        private void HandlePinMarkerClick(IPinMarker? newPinMarker)
+        {
+            if (newPinMarker != previousClickedMarker)
+            {
+                previousClickedMarker?.AnimateDeselectionAsync().Forget();
+                newPinMarker?.AnimateSelectionAsync().Forget();

```

Comment:
Question: shouldn't we manage cancellation tokens for these async calls?

Reply:
Good point!! we probably should